managed/services/supervisord/logs.go

```diff
@@ -302,13 +302,18 @@ func readLog(name string, maxLines int, maxBytes int64) ([]byte, time.Time, erro
 	}
 
 	r := ring.New(maxLines)
-	s := bufio.NewScanner(f)
-	for s.Scan() {
-		r.Value = []byte(s.Text() + "\n")
+	reader := bufio.NewReader(f)
+	for {
+		b, err := reader.ReadBytes('\n')
+		if err == io.EOF {
+			break
+		}
```

Comment:
Looks like we're losing the last part if source doen't end with new line.

Reply:
Good catch